File: api/routes/agent.py

from fastapi import APIRouter
from api.schemas.agent import AgentRequest
from agent.runtime import run_agent_execution_debug

router = APIRouter()

@router.post("/agent/run")
async def run_agent(request: AgentRequest):
  query = request.query
  result = await run_agent_execution_debug(query)
  return result

File: nextjs/services/agent.ts
import { api } from "@/lib/api";
import type { AgentRequest, AgentResponse } from "@/types/agent";

export async function runAgent(request: AgentRequest): Promise<AgentResponse> {
  const response = await api.post<AgentResponse>("/agent/run", request);
  return response.data;
}
